crates/worker/src/partition/state_machine/effects.rs

```diff
@@ -955,36 +980,44 @@ impl Effects {
 
     /// We log only if the log level is TRACE, or if the log level is DEBUG and we're the leader,
     /// or if the span level is INFO and we're the leader.
-    pub(crate) fn log(
-        &self,
-        is_leader: bool,
-        full_invocation_id: Option<FullInvocationId>,
-        related_span: SpanRelation,
-    ) {
+    pub(crate) fn log(&self, is_leader: bool) {
         // Skip this method altogether if logging is disabled
         if !(((event_enabled!(Level::DEBUG) || span_enabled!(Level::INFO)) && is_leader)
             || event_enabled!(Level::TRACE))
         {
             return;
         }
 
-        let span = match (is_leader, full_invocation_id) {
-            (true, Some(full_invocation_id)) => debug_span!(
-                "state_machine_effects",
-                rpc.service = %full_invocation_id.service_id.service_name,
-                restate.invocation.id = %full_invocation_id,
-            ),
-            (false, Some(full_invocation_id)) => trace_span!(
-                "state_machine_effects",
-                rpc.service = %full_invocation_id.service_id.service_name,
-                restate.invocation.id = %full_invocation_id,
-            ),
-            (true, None) => debug_span!("state_machine_effects"),
-            (false, None) => trace_span!("state_machine_effects"),
+        let span = if is_leader {
+            debug_span!("state_machine_effects")
+        } else {
+            trace_span!("state_machine_effects")
         };
 
+        if let Some(invocation_id) = &self.related_invocation_id {
+            span.record(
+                "restate.invocation.id",
+                tracing::field::display(invocation_id),
+            );
+        }
+
+        if let Some(invocation_target) = &self.related_invocation_target {
+            span.record(
+                "rpc.service",
+                tracing::field::display(invocation_target.service_name()),
+            );
+            span.record(
+                "rpc.method",
+                tracing::field::display(invocation_target.handler_name()),
+            );
+            span.record(
+                "restate.invocation.target",
+                tracing::field::display(invocation_target),
+            );
+        }
+
         // Create span and enter it
-        related_span.attach_to_span(&span);
+        self.related_span.clone().attach_to_span(&span);
```

Comment:
Unclear why do we need to clone here.

Reply:
`attach_to_span()` takes self by value, let me change that.